packages/plugins/ganache/src/index.js

```diff
@@ -1,10 +1,41 @@
+const constants = require('embark-core/constants');
+
 class Ganache {
   constructor(embark) {
-    embark.events.request('blockchain:vm:register', () => {
+    embark.events.request("blockchain:node:register", constants.blockchain.clients.ganache, {
+      isStartedFn: (cb) => {
+        cb(null, true); // Always assume it's started, because it's just a provider (nothing to start)
+      },
+      launchFn: (cb) => {
+        cb(); // Will probably never be called, but if so, callback straight aways because there is nothing to start
+      },
+      stopFn: async (cb) => {
+        cb(); // Will probably never be called, but if so, callback straight aways because there is nothing to start
+      }
+    });
+
+    embark.events.request('blockchain:vm:register', constants.blockchain.clients.ganache, () => {
       const ganache = require('ganache-cli');
-      // Default to 8000000, which is the server default
-      // Somehow, the provider default is 6721975
-      return ganache.provider({gasLimit: '0x7A1200'});
+      const blockchainConfig = embark.config.blockchainConfig;
+
+      // Ensure the dir exists before initiating Ganache, because Ganache has a bug
+      // => https://github.com/trufflesuite/ganache-cli/issues/558
+      embark.fs.ensureDirSync(blockchainConfig.datadir);
+
+      const hasAccounts = blockchainConfig.accounts && blockchainConfig.accounts.length;
+
+      const isTest = embark.currentContext.includes('test');
+
+      return ganache.provider({
+        // Default to 8000000, which is the server default
+        // Somehow, the provider default is 6721975
+        gasLimit: blockchainConfig.targetGasLimit || '0x7A1200',
+        blockTime: blockchainConfig.simulatorBlocktime,
+        network_id:  blockchainConfig.networkId || 1337,
+        db_path: isTest ? '' : blockchainConfig.datadir,
+        default_balance_ether: '99999',
+        mnemonic: hasAccounts || isTest ? '' : 'example exile argue silk regular smile grass bomb merge arm assist farm'
```

Comment:
What about if the user doesn't want to define any accounts in the config (ie maybe just rely on metamask), is that possible?

Reply:
> `maybe just rely on metamask, is that possible?`

yep. the user can always use whatever account they wants in metamask independently of what's configured.  this mnemonic here only affects any config that would uses a `nodeAccount`.

@jrainville I think we could make the mnemonic configurable in the client specific options of blockchainConfig

Reply:
i.e `blockchainConfig.clientConfig.mnemonic`

Reply:
If a user doesn't specify one should we generate a random one instead of having a fixed default?  I don't think there's a huge security issue, but it seems like it would be "more right" somehow.

If we stick with a fixed default, can we move it into `constants.blockchain`?

Reply:
> yep. the user can always use whatever account they wants in metamask independently of what's configured. this mnemonic here only affects any config that would uses a nodeAccount.

My point here is what the user explicitly doesn't want any accounts? It doesn't seem possible due to this line, as a default mnemonic will always be used.

Reply:
> My point here is what if the user explicitly doesn't want any accounts? It doesn't seem possible due to this line, as a default mnemonic will always be used.

Maybe test if `blockchainConfig.clientConfig.mnemonic === false`?

Reply:
This doesn't affect Metamask usage at all. If the user uses Metamask, it means it connects using `$WEB3`, so in never connect to the proxy, nor to the node/vm.

As for the reason I put default accounts/default mnemomnic, it's so it mimics the simulator feel (it's the same mnemonic too) and also, just like Geth's `--dev` mode, it needs to at least have one default account so that everything works out of the box.

I don't mind using a random mnemonic, I think it's doable with Ganache with one of the property.

Reply:
So, I just tested and having different accounts each time is not very useful and even causes issues with Ganache's datadir, because having different accounts each time makes it not keep track of the last deployment and just have a bug (weird Ganache error).

Reply:
Since a unique mnemonic each time isn't really viable, as you described, could we move the fixed one into `blockchain.constants`?